videostore/videostore.go

```diff
@@ -497,3 +518,79 @@ func (vs *videostore) Close() {
 		}
 	}
 }
+
+// GetStorageState returns the current storage state for this VideoStore.
+func (vs *videostore) GetStorageState() (*StorageState, error) {
+	storagePath := vs.config.Storage.StoragePath
+	usedBytes, err := getDirectorySize(storagePath)
+	if err != nil {
+		return nil, err
+	}
+	usedGB := float64(usedBytes) / float64(gigabyte)
+	limitGB := float64(vs.config.Storage.SizeGB)
+
+	fsUsage, err := diskusage.Statfs(storagePath)
+	if err != nil {
+		return nil, err
+	}
+	remainingGB := float64(fsUsage.AvailableBytes) / float64(gigabyte)
+
+	files, err := getSortedFiles(storagePath)
+	if err != nil {
+		return nil, err
+	}
+
+	var (
+		ranges                 []VideoTimeRange
+		currentFrom, currentTo time.Time
+		seenFirstSegment       bool
+	)
+
+	for _, file := range files {
+		info, err := getVideoInfo(file.name)
+		if err != nil {
+			vs.logger.Warnf("skipping unfinished or corrupt file %s: %v", file.name, err)
+			continue
+		}
+
+		start := file.startTime
+		end := start.Add(info.duration)
+
+		if !seenFirstSegment {
+			currentFrom, currentTo = start, end
+			seenFirstSegment = true
+			continue
+		}
+
+		if !start.After(currentTo) {
```

Comment:
Is this essentially a 1 second buffer? Or this expected to hit on every new segment?

Reply:
I expect this to trigger a bunch of false positives given if there is even a few milliseconds missing between one segement file and another we are going to create a new `VideoTimeRange`

I recommend providing a little bit of slop (lets say 1 second) that can be missing from one file starting and another file ending and we can still consider the video segments contiguous.